### PR TITLE
feat: make endpoint to create ratings

### PR DESCRIPTION
**Pull request by anand2312**

### CODE REVIEW

File: src/backend/routers/post.py [hunk 1]
```diff
@@ -179,6 +184,25 @@ async def get_comments(
     )
 
 
+@router.post("/{post_id}/rating")
```
Comment (vivekashok1221) [inline src/backend/routers/post.py:187]: NIT: `post_id` should be `id`  for consistency.
Small disciplines repeated with consistency everyday lead to great achievements gained slowly over time.

Comment (vivekashok1221) [reply src/backend/routers/post.py:187]: Hmm, in some place it's `post_id` whereas in others  it's `id`. We should make it consistent throughout. An inconsistent mind is disorganised and undisciplined.

I prefer `post_id` over `id` btw.

File: src/backend/routers/post.py [hunk 2]
```diff
@@ -179,6 +184,25 @@ async def get_comments(
...
+        rating_record = await PostRating.prisma().upsert(
+            where={"post_id_author_id": {"post_id": post_id, "author_id": str(user_id)}},
+            data={
+                "create": {"post_id": post_id, "author_id": str(user_id), "value": rating.rating},
+                "update": {"value": rating.rating},
+            },
+        )
+    except PrismaError as e:
+        logger.warning(f"Could not create rating: {e}")
+        raise HTTPException(400, "Could not create the rating due to an internal error")
```
Comment (vivekashok1221) [inline src/backend/routers/post.py:202]: Internal errors should be 500, ya.
If you can think of any specific prisma errors, raise  4xx error codes in those cases and throw 5xx in the general case.

Comment (vivekashok1221) [inline src/backend/routers/post.py:202]: Internal errors should be 500, ya.
If you can think of any specific prisma errors, raise  4xx error codes in those cases and throw 5xx in the general case.

**Review by vivekashok1221 — CHANGES_REQUESTED** [review]
Untested but code looks functional. I will test once I get a chance.

File: src/backend/routers/post.py (hunk 1 above)
Comment (anand2312) [reply src/backend/routers/post.py:187]: I just copied the function for create comment here lmao. I can change the rest of the functions to use `post_id` if you want.

Comment (vivekashok1221) [reply src/backend/routers/post.py:187]: I want. I need.

File: src/backend/routers/post.py (hunk 2 above)
Comment (anand2312) [reply src/backend/routers/post.py:202]: Resolved in [80fbdaa](https://github.com/idkbrowby/gamingrealm-backend/commit/80fbdaa2f098ea80a223baadab0d284b1f950e27)

File: src/backend/routers/post.py (hunk 1 above)
Comment (anand2312) [reply src/backend/routers/post.py:187]: Resolved in [fdff730](https://github.com/idkbrowby/gamingrealm-backend/commit/fdff7300c23bee59889690e6d59e161f9c4c59b6)

File: src/backend/routers/post.py [hunk 3]
```diff
@@ -94,11 +96,12 @@ async def search_posts(
...
+@router.get("/{post_id}")
+async def get_post(post_id: str) -> PostDetails:
     """Get full details of a specific post."""
     post = await Post.prisma().find_first(
-        where={"id": id, "deleted": False}, include={"tags": True, "media": True, "author": True}
+        where={"id": post_id, "deleted": False},
+        include={"tags": True, "media": True, "author": True},
     )
     if not post:
         raise HTTPException(404, "Post not found")
```
Comment (vivekashok1221) [inline src/backend/routers/post.py:107]: `id` needs to be changed to `post_id` on [line 111](https://github.com/idkbrowby/gamingrealm-backend/blob/anand/rating-endpoints/src/backend/routers/post.py#L111)

**Review by vivekashok1221 — CHANGES_REQUESTED** [review]

**Review by vivekashok1221 — APPROVED** [review]
lgtm

**Pull request merged**